src/components/SearchBox.tsx

```diff
@@ -42,10 +42,18 @@ const SearchBox = ({ setSearchResults }: SearchBoxProps) => {
       } = await axios.get(
         API_URL + `?term=${searchValue}&entity=album&limit=10`
       );
-      setSearchResults(results);
-      setKeyword(searchValue);
+
+      if (results.length === 0) {
+        setHelperText('No results match criteria.');
+      } else {
+        setSearchResults(results);
+        setKeyword(searchValue);
+        setHelperText('');
+      }
     } catch (error) {
+      console.log(error);
       setHelperText('No results match criteria.');
```

Comment:
The approach is good but, as you can see, now the situation with no results and situation where we have an api error share the same message. This is an example of sub-par user experience. In case of an error, we can keep the console log but i would like to rephrase the error message into: Something went wrong. Please try again.